src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseAssignedUserRolesController.java

```diff
@@ -0,0 +1,107 @@
+    private final CaseAssignedUserRolesOperation caseAssignedUserRolesOperation;
+
+    @Autowired
+    public CaseAssignedUserRolesController(UIDService caseReferenceService,
+                                           @Qualifier("authorised") CaseAssignedUserRolesOperation caseAssignedUserRolesOperation) {
+        this.caseReferenceService = caseReferenceService;
+        this.caseAssignedUserRolesOperation = caseAssignedUserRolesOperation;
+    }
+
+    @GetMapping(
+        path = "/case-users",
+        produces = {
+            V2.MediaType.CASE_USER_ROLES
+        }
+    )
+    @ApiOperation(
+        value = "Retrieve Case user roles by Case ID and User ID"
+    )
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
+            message = "Case-User-Role assignments returned successfully",
+            response = CaseAssignedUserRolesResource.class
+        ),
+        @ApiResponse(
+            code = 400,
+            message = V2.Error.CASE_ID_INVALID
+        ),
+        @ApiResponse(
+            code = 400,
+            message = V2.Error.EMPTY_CASE_ID_LIST
+        ),
+        @ApiResponse(
+            code = 400,
+            message = V2.Error.USER_ID_INVALID
+        ),
+        @ApiResponse(
+            code = 403,
+            message = V2.Error.OTHER_USER_CASE_ROLE_ACCESS_NOT_GRANTED
+        )
+    })
+    // TODO: 01/06/2020  add LogAudit
+    public ResponseEntity<CaseAssignedUserRolesResource> getCaseUserRoles(@RequestParam("case_ids") String caseIds,
+                                                                          @RequestParam(value = "user_ids", required = false) String userIds) {
+        List<String> listUserIds = convertToList(userIds);
+        List<String> listCaseIds = convertToList(caseIds);
+        validateRequestParams(listCaseIds, listUserIds);
+
+        List<CaseAssignedUserRole> caseAssignedUserRoles = this.caseAssignedUserRolesOperation.findCaseUserRoles(listCaseIds
+            .stream()
+            .map(caseId -> Long.valueOf(caseId))
+            .collect(Collectors.toCollection(ArrayList::new)), listUserIds);
+        return ResponseEntity.ok(new CaseAssignedUserRolesResource(caseIds, userIds, caseAssignedUserRoles));
+    }
+
+    private List<String> convertToList(String ids) {
+        return ids == null ? Collections.EMPTY_LIST : Arrays.asList(ids.split(","));
+    }
+
+    private void validateRequestParams(List<String> caseIds, List<String> userIds) {
+        if (caseIds.isEmpty()) {
+            throw new BadRequestException(V2.Error.EMPTY_CASE_ID_LIST);
+        }
+
+        caseIds.forEach(caseId -> {
+            if (!caseReferenceService.validateUID(caseId)) {
+                throw new BadRequestException(V2.Error.CASE_ID_INVALID);
+            }
+        });
+
+        userIds.forEach(userId -> {
+            if (StringUtils.isAllBlank(userId)) {
+                throw new BadRequestException(V2.Error.USER_ID_INVALID);
+            }
+        });
+    }
```

Comment:
This throws an Exception upon the first issue it encounters. However, all validation issues should be reported at once.